support-frontend/app/services/stepfunctions/Client.scala

```diff
@@ -34,11 +34,17 @@ class Client(client: AWSStepFunctionsAsync, arn: StateMachineArn) extends SafeLo
   private def startExecution(arn: String, input: String, name: String)(implicit
       ec: ExecutionContext,
   ): Response[StartExecutionResult] = convertErrors {
+    val proposedName = name + "-" + System.nanoTime().toString
+    val validName =
+      if (proposedName.length <= 80)
+        proposedName
+      else
+        (System.nanoTime().toString + "-" + name).take(78) + "--"
```

Comment:
To avoid having the prefix changing, could we use:
```suggestion
        // maxLength - System.nanoTime().toString.size - truncationString.size
        name.take(80 - 16 - 2) + System.nanoTime().toString + "--"
```

Reply:
yeah good idea I'd like that too, i just wanted to prioritise reliability over consistency (although test users are going to be affected before real ones)

Reply:
actually MAX and MIN LONG are 19 and 20 characters respectively when run through toString.  Safer to toString the value first, measure it, and check the total.
```
123456789 123456789 123456789
9223372036854775807
-9223372036854775808
```

Reply:
OK I have pushed updates so it takes (exactly) 11 characters for the ID while remaining valid for step functions, and some tests to make sure, so the overflow issue has gone away for now.  Please have a look and see what you think @jamesgorrie